### ISSUE TITLE
[Test Case Results] Essorcal

### ISSUE THREAD

**Issue by Essorcal** [rlguy/Blender-FLIP-Fluids-Beta#96]
I ran the cascading water test case at 100 resolution in 14:10.  Seems like it worked to me. Definitely didn't seem to be using much CPU or any GPU. Results below:

[test_case_cascading_water_feature100.zip](https://github.com/rlguy/Blender-FLIP-Fluids-Beta/files/1749124/test_case_cascading_water_feature100.zip)

**Comment by rlguy** [comment]
Thanks Essorcal!

I checked your results and it looks like the test ran okay, but there was an error preventing your GPU from being used. OpenCL failed initialization with an 'CL_OUT_OF_RESOURCES' error. This will decrease average CPU usage and GPU usage. I will look into this.

This issue does not break the simulator, so you will still be able to receive and test builds. I have updated your Dropbox folder with a build of the addon for testing. Your beta invite email should contain a link and a password to your Dropbox folder.

I will be closing this issue. Please post any further test case results to this thread.

**Comment by rlguy** [comment]
I have checked your system info and it looks like Cycles was also having trouble attempting to initialize your GPU device. So perhaps there could be a problem with your NVIDIA drivers?

```
Cycles:
===========================================

CPU device capabilities: SSE2 SSE3 SSE41 AVX AVX2

OpenCL device capabilities:
Number of platforms: 1
Platform #0
	Platform Name: NVIDIA CUDA
	Platform Vendor: NVIDIA Corporation
	Platform Version: OpenCL 1.2 CUDA 9.1.75
	Platform Profile: FULL_PROFILE
	Platform Extensions: cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64 cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_d3d10_sharing cl_khr_d3d10_sharing cl_nv_d3d11_sharing cl_nv_copy_opts cl_khr_gl_event cl_nv_create_buffer
	Number of devices: 1
		Device: #0
			Device Name: Quadro M6000 24GB
			Device Vendor: NVIDIA Corporation
			Device OpenCL C Version: OpenCL C 1.2 
			Device Profile: FULL_PROFILE
			Device Version: OpenCL 1.2 CUDA
			Device Extensions: cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64 cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_d3d10_sharing cl_khr_d3d10_sharing cl_nv_d3d11_sharing cl_nv_copy_opts cl_khr_gl_event cl_nv_create_buffer

CUDA device capabilities:
Error initializing CUDA: Unknown error
Enabled add-ons:
===========================================
```

**Comment by Essorcal** [comment]
I did a clean install of NVIDIA Quadro Drivers 388.16 and re-ran the test case.  Another issue i have is the completed frames section of the simulation stats stops updating randomly.  This happened with the original case as well.  For this one it was at 1.8% the timing stats change occasionally but the only way I could tell it was finished was waiting 10-15 minutes then pressing the stop/pause button, having it be stuck at "pausing/cancelling safe to close blender" saving the file and then closing and re-opening blender where i get the resume/reset completed option

[test_case_cascading_water_feature100.zip](https://github.com/rlguy/Blender-FLIP-Fluids-Beta/files/1751968/test_case_cascading_water_feature100.zip)

**Comment by rlguy** [comment]
Thanks for updating me! It looks like re-installing the drivers fixed the OpenCL issue. Both the simulator and Cycles are able to recognised your GPU device without errors now.

The stats update stopping sounds like a bug. Let me know if this happens again, and if it does please let me know if there are any errors displayed in the Blender system console. And I am wondering:

- Is your cache directory located on a network drive?
- Is your cache directory located in a Dropbox folder?

**Comment by Essorcal** [comment]
So I think I fixed the issue it was a cache problem and it was resolved when i moved the cache from an internal spinning disk storage drive to the SSD where blender is installed. Below is what I wrote before changing the cache location if it's helpful for tracking anything down.

Cleared and re-ran a few more times. It happens every time at the 100 resolution in the first 20 or so frames, I ran a test at the 400 resolution level and it did not happen in the first 30 or so.  I don't know if its updating too quickly and losing track of the value but thats my initial observation. When i get a chance to dedicate time to it i'll run a full length test at the 400 resolution and level to see if it happens again.  The cache is not located on a network drive or dropbox folder however it is on a separate internal hard drive than my blender installation.
